File: app/src/main/java/com/ben/bencustomerserver/listener/OnRecallMessageResultListener.java
package com.ben.bencustomerserver.listener;


import com.ben.bencustomerserver.model.BaseMessageModel;

public interface OnRecallMessageResultListener {
    /**
     * Recall successful
     * @param originalMessage The message was unsent
     * @param notification  The notification message
     */
    void recallSuccess(BaseMessageModel originalMessage, BaseMessageModel notification);

    /**
     * Recall failed
     * @param code
     * @param errorMsg
     */
    void recallFail(int code, String errorMsg);
}

File: lib-bencs/src/main/java/com/ben/bencustomerserver/listener/OnRecallMessageResultListener.kt
package com.ben.bencustomerserver.listener

import com.ben.bencustomerserver.model.BaseMessageModel

interface OnRecallMessageResultListener {
    /**
     * Recall successful
     * @param originalMessage The message was unsent
     * @param notification  The notification message
     */
    fun recallSuccess(originalMessage: BaseMessageModel?, notification: BaseMessageModel?)

    /**
     * Recall failed
     * @param code
     * @param errorMsg
     */
    fun recallFail(code: Int, errorMsg: String?)
}